### PR TITLE
WebGL renderer / use the specified loading strategy for the vector data

### LINKED ISSUE #10173: Heatmap issue with GeoJSON and loadingstrategy as tile
Hi 

I am trying to create a Heatmap with a GeoJSON source and loadingstrategy.tile. I am running the code below and openlayers is not displaying anything or crashes the browser.

var source = new ol.source.Vector({
                            format: new ol.format.GeoJSON({
                                dataProjection: projectionCode,
                                geometryName: geometryName
                            }),
                            loader: function(...),
                            strategy: ol.loadingstrategy.tile(new ol.tilegrid.createXYZ({extent: getExtent() })),
                            projection: projectionCode
                        });

var vector = new ol.layer.Heatmap({
              source: source,
              blur: 25,
              radius: 15
            });

vector.getSource().on('addfeature', function(event) {
              event.feature.set('weight', 1);
});

I managed to get round the problem by using loadingstrategy.tile with a ol.layer.Vector and using loader function to fill Heatmap source.

### PR DESCRIPTION
**Pull request by jahow**
Previously the WebGL points renderer would specify an infinite extent when loading features, bypassing the loading strategy and potentially crashing the browser. It now specifies the current view extent to `vectorSource.loadFeatures`.

Fixes #10173 

@mike-000 could you please take a look? thanks!

### CODE REVIEW

**Comment by jahow** [comment]
I have updated the PR to fix the behaviour of the renderer. Now the features loading is called when:
* either the source revision has changed, or the current extent
* the view is not moving

Previously the source was only reloaded when its revision was changed. New tests were added to reflect that. Thanks for any review!

**Comment by mike-000** [comment]
With the latest change it looks good (the calls to the loader are matching those of the normal vector layer).

File: src/ol/renderer/webgl/PointsLayer.js [hunk 1]
```diff
@@ -296,20 +297,22 @@ class WebGLPointsLayerRenderer extends WebGLLayerRenderer {
...
+    if (viewNotMoving && (extentChanged || sourceChanged)) {
       const projection = viewState.projection;
       const resolution = viewState.resolution;
-      vectorSource.loadFeatures([-Infinity, -Infinity, Infinity, Infinity], resolution, projection);
-    }
 
-    const viewNotMoving = !frameState.viewHints[ViewHint.ANIMATING] && !frameState.viewHints[ViewHint.INTERACTING];
-    const extentChanged = !equals(this.previousExtent_, frameState.extent);
-    if ((sourceChanged || extentChanged) && viewNotMoving) {
+      const renderBuffer = layer instanceof BaseVector ? layer.getRenderBuffer() : 0;
```
Comment (fredj) [inline src/ol/renderer/webgl/PointsLayer.js]: did you use `instanceof` because of an error from the type checker?

Comment (jahow) [reply src/ol/renderer/webgl/PointsLayer.js]: Actually the renderer can use layers that do not inherit from the `BaseVector` class, as seen here: https://github.com/openlayers/openlayers/blob/master/src/ol/layer/WebGLPoints.js

The only requirement is having a vector source.

Comment (fredj) [reply src/ol/renderer/webgl/PointsLayer.js]: ok, got it now.

**Comment by jahow** [comment]
Thanks to both for the reviews!

**Pull request merged**